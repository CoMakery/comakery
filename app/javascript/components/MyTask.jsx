import React from 'react'
import PropTypes from 'prop-types'
import CurrencyAmount from './CurrencyAmount'
import Userpics from './Userpics'
import styled, { css } from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  box-shadow: 0 5px 10px 0 rgba(0, 0, 0, 0.1);
  height: 80px;
  padding: 10px 20px;
  margin-bottom: 10px;
  text-transform: uppercase;
`

const RightBorder = styled.div`
  width: 2px;
  height: 100px;
  box-shadow: 0 5px 10px 0 rgba(0, 0, 0, .2);
  background-color: #5037f7;
  z-index: 10;
  position: absolute;
  margin-left: -21px;
  margin-top: -10px;

  ${props => props.status === 'ready' && css`
    background-color: #4a4a4a;
  `}

  ${props => props.status === 'started' && css`
    background-color: #008e9b;
  `}

  ${props => props.status === 'submitted' && css`
    background-color: #007ae7;
  `}

  ${props => props.status === 'accepted' && css`
    background-color: #5037f7;
  `}

  ${props => props.status === 'paid' && css`
    background-color: #fb40e5;
  `}

  ${props => props.status === 'rejected' && css`
    background-color: #ff4d4d;
  `}
`

const FirstRow = styled.div`
  margin-bottom: 10px;
  display: flex;
  width: 100%;
  justify-content: space-between;
  align-items: flex-start;
`

const SecondRow = styled.div`
  min-height: 28px;
  margin-bottom: 12px;
  display: flex;
  width: 100%;
  justify-content: space-between;
`

const ThirdRow = styled.div`
  display: flex;
  width: 100%;
`

const Name = styled.div`
  font-family: Montserrat;
  font-size: 16px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #3a3a3a;
`

const BlockWrapper = styled.div`
  margin-right: auto;
`

const Mission = styled.div`
  font-family: Montserrat;
  font-size: 12px;
  font-weight: 600;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.4;
  letter-spacing: normal;
  color: #4a4a4a;
  text-transform: none;

  a {
    text-decoration: none;
    color: #0089f4;

    &:hover {
      text-decoration: underline;
    }
  }
`

const Project = styled.div`
  font-family: Montserrat;
  font-size: 12px;
  font-weight: 600;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.4;
  letter-spacing: normal;
  color: #4a4a4a;
  text-transform: none;

  a {
    text-decoration: none;
    color: #0089f4;

    &:hover {
      text-decoration: underline;
    }
  }
`

const Status = styled.div`
  font-family: Montserrat;
  font-size: 10px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  margin-right: 2em;

  b {
    font-weight: 900;
  }
`

const Type = styled.div`
  font-family: Montserrat;
  font-size: 10px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  margin-right: 2em;

  b {
    font-weight: 900;
  }
`

const Contributor = styled.div`
  font-family: Montserrat;
  font-size: 10px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  flex-wrap: nowrap;
  display: flex;
  margin-right: 2em;

  b {
    font-weight: 900;
  }
`

class MyTask extends React.Component {
  render() {
    let task = this.props.task
    return (
      <React.Fragment>
        <Wrapper>
          <RightBorder status={task.status} />
          <FirstRow>
            <Name>{task.name}</Name>
            <CurrencyAmount
              amount={task.totalAmount}
              currency={task.token.currency}
              logoUrl={task.token.logo}
            />
          </FirstRow>

          <SecondRow>
            <BlockWrapper>
              <Project>PROJECT: <a href={task.project.url}>{task.project.name}</a></Project>
              {task.mission.name &&
                <Mission>MISSION: <a href={task.mission.url}>{task.mission.name}</a></Mission>
              }
            </BlockWrapper>
          </SecondRow>

          <ThirdRow>
            <Status>
              <b>{task.status} </b>
              {task.updatedAt} ago
            </Status>
            <Type>
              <b>TYPE </b>
              {task.batch.specialty || 'General'}
            </Type>
            {task.contributor.name &&
              <Contributor>
                <Userpics pics={[task.contributor.image]} limit={1} />
                {task.contributor.name}
              </Contributor>
            }
          </ThirdRow>
        </Wrapper>
      </React.Fragment>
    )
  }
}

MyTask.propTypes = {
  task: PropTypes.object
}
MyTask.defaultProps = {
  task: {
    status: null,
    token : {
      currency: 'test',
      logo    : 'test'
    },
    project: {
      name: null,
      url : null
    },
    mission: {
      name: null,
      url : null
    },
    batch: {
      specialty: null
    },
    contributor: {
      name : null,
      image: null
    }
  }
}
export default MyTask
