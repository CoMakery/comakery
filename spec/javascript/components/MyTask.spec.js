import React from 'react'
import { mount } from 'enzyme'
import MyTask from 'components/MyTask'

describe('MyTask', () => {
  it('renders component correctly', () => {
    const wrapper = mount(<MyTask />)

    expect(wrapper).toMatchSnapshot()
  })
})
