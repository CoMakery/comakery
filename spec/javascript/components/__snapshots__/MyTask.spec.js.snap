// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyTask renders component correctly 1`] = `
<MyTask
  task={
    Object {
      "batch": Object {
        "specialty": null,
      },
      "contributor": Object {
        "image": null,
        "name": null,
      },
      "mission": Object {
        "name": null,
        "url": null,
      },
      "project": Object {
        "name": null,
        "url": null,
      },
      "status": null,
      "token": Object {
        "currency": "test",
        "logo": "test",
      },
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htoDjs",
            "isStatic": true,
            "lastClassName": "fRwkQd",
            "rules": Array [
              "
  display: flex;
  flex-direction: column;
  box-shadow: 0 5px 10px 0 rgba(0, 0, 0, 0.1);
  height: 80px;
  padding: 10px 20px;
  margin-bottom: 10px;
  text-transform: uppercase;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htoDjs",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-htoDjs fRwkQd"
      >
        <styled.div
          status={null}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-dnqmqq",
                  "isStatic": false,
                  "lastClassName": "SIvvw",
                  "rules": Array [
                    "
  width: 2px;
  height: 100px;
  box-shadow: 0 5px 10px 0 rgba(0, 0, 0, .2);
  background-color: #5037f7;
  z-index: 10;
  position: absolute;
  margin-left: -21px;
  margin-top: -10px;

  ",
                    [Function],
                    "

  ",
                    [Function],
                    "

  ",
                    [Function],
                    "

  ",
                    [Function],
                    "

  ",
                    [Function],
                    "

  ",
                    [Function],
                    "
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-dnqmqq",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            status={null}
          >
            <div
              className="sc-dnqmqq SIvvw"
            />
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-iwsKbI",
                  "isStatic": true,
                  "lastClassName": "enWBVh",
                  "rules": Array [
                    "
  margin-bottom: 10px;
  display: flex;
  width: 100%;
  justify-content: space-between;
  align-items: flex-start;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-iwsKbI",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-iwsKbI enWBVh"
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-VigVT",
                        "isStatic": true,
                        "lastClassName": "jsKpKw",
                        "rules": Array [
                          "
  font-family: Montserrat;
  font-size: 16px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #3a3a3a;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-VigVT",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-VigVT jsKpKw"
                  />
                </StyledComponent>
              </styled.div>
              <CurrencyAmount
                amount=""
                currency="test"
                logoUrl="test"
              >
                <styled.div>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bdVaJa",
                          "isStatic": true,
                          "lastClassName": "khtMgg",
                          "rules": Array [
                            "
  display: flex;
  flex-direction: row;
  align-items: center;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: bold;
  color: #201662;
  padding-right: 15px;
",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bdVaJa",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="sc-bdVaJa khtMgg"
                    >
                      <styled.div>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bwzfXH",
                                "isStatic": true,
                                "lastClassName": "eNKWhX",
                                "rules": Array [
                                  "
  margin-right: 0.5em;
",
                                ],
                              },
                              "displayName": "styled.div",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bwzfXH",
                              "target": "div",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <div
                            className="sc-bwzfXH eNKWhX"
                          />
                        </StyledComponent>
                      </styled.div>
                      <styled.div>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-htpNat",
                                "isStatic": true,
                                "lastClassName": "eglqcF",
                                "rules": Array [
                                  "
  margin-right: 0.5em;
  text-transform: uppercase;
",
                                ],
                              },
                              "displayName": "styled.div",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-htpNat",
                              "target": "div",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <div
                            className="sc-htpNat eglqcF"
                          >
                            test
                          </div>
                        </StyledComponent>
                      </styled.div>
                      <styled.img
                        alt="test logo"
                        src="test"
                      >
                        <StyledComponent
                          alt="test logo"
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bxivhb",
                                "isStatic": true,
                                "lastClassName": "iJrLVE",
                                "rules": Array [
                                  "
  height: 14px;
",
                                ],
                              },
                              "displayName": "styled.img",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bxivhb",
                              "target": "img",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          src="test"
                        >
                          <img
                            alt="test logo"
                            className="sc-bxivhb iJrLVE"
                            src="test"
                          />
                        </StyledComponent>
                      </styled.img>
                    </div>
                  </StyledComponent>
                </styled.div>
              </CurrencyAmount>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-gZMcBi",
                  "isStatic": true,
                  "lastClassName": "cMjsy",
                  "rules": Array [
                    "
  min-height: 28px;
  margin-bottom: 12px;
  display: flex;
  width: 100%;
  justify-content: space-between;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-gZMcBi",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-gZMcBi cMjsy"
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-jTzLTM",
                        "isStatic": true,
                        "lastClassName": "liyEvD",
                        "rules": Array [
                          "
  margin-right: auto;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-jTzLTM",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-jTzLTM liyEvD"
                  >
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-jzJRlG",
                              "isStatic": true,
                              "lastClassName": "DVEmp",
                              "rules": Array [
                                "
  font-family: Montserrat;
  font-size: 12px;
  font-weight: 600;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.4;
  letter-spacing: normal;
  color: #4a4a4a;
  text-transform: none;

  a {
    text-decoration: none;
    color: #0089f4;

    &:hover {
      text-decoration: underline;
    }
  }
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-jzJRlG",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-jzJRlG DVEmp"
                        >
                          PROJECT: 
                          <a
                            href={null}
                          />
                        </div>
                      </StyledComponent>
                    </styled.div>
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-gqjmRU",
                  "isStatic": true,
                  "lastClassName": "kpfwRn",
                  "rules": Array [
                    "
  display: flex;
  width: 100%;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-gqjmRU",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-gqjmRU kpfwRn"
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-cSHVUG",
                        "isStatic": true,
                        "lastClassName": "EQMTJ",
                        "rules": Array [
                          "
  font-family: Montserrat;
  font-size: 10px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  margin-right: 2em;

  b {
    font-weight: 900;
  }
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-cSHVUG",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-cSHVUG EQMTJ"
                  >
                    <b>
                       
                    </b>
                     ago
                  </div>
                </StyledComponent>
              </styled.div>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-kAzzGY",
                        "isStatic": true,
                        "lastClassName": "jYrEjD",
                        "rules": Array [
                          "
  font-family: Montserrat;
  font-size: 10px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  margin-right: 2em;

  b {
    font-weight: 900;
  }
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-kAzzGY",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-kAzzGY jYrEjD"
                  >
                    <b>
                      TYPE 
                    </b>
                    General
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</MyTask>
`;
